# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: Cargo.toml
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,30 +25,30 @@ exclude = [
   "build/liblowercase",
 
   # Exclude third-party code vendored into mozilla-central.
   "servo",
   "third_party/rust",
 
   # Excluded because these crates have their own Cargo workspaces so they can't
   # be included in the top-level one.
-  "gfx/wgpu",
   "gfx/wr",
 
   # Excluded because they are used only as dependencies, not top-level targets,
   # so we don't need to vendor their dev-dependencies.
   "gfx/webrender_bindings",
   "media/mp4parse-rust/mp4parse",
   "media/mp4parse-rust/mp4parse_capi",
   "xpcom/rust/gkrust_utils",
   "tools/lint/test/files/clippy",
   "tools/fuzzing/rust",
 ]
 
 # Use the new dependency resolver to reduce some of the platform-specific dependencies.
+# This is required for 'third_party/rust/wgpu-hal'
 resolver = "2"
 
 # Explicitly specify what our profiles use.  The opt-level setting here is
 # a total fiction; see the setup of MOZ_RUST_DEFAULT_FLAGS for what the
 # opt-level setting will be as a result of various other configure flags.
 [profile.dev]
 opt-level = 1
 rpath = false
@@ -77,18 +77,23 @@ opt-level = 2
 [profile.release.package.glsl]
 opt-level = 2
 
 [patch.crates-io]
 chardetng = { git = "https://github.com/hsivonen/chardetng", rev="302c995f91f44cf26e77dc4758ad56c3ff0153ad" }
 chardetng_c = { git = "https://github.com/hsivonen/chardetng_c", rev="ed8a4c6f900a90d4dbc1d64b856e61490a1c3570" }
 libudev-sys = { path = "dom/webauthn/libudev-sys" }
 packed_simd = { git = "https://github.com/hsivonen/packed_simd", rev="6a16f954950401b92b4e220fbf2dfaf6f00e1fb2" }
-spirv_cross = { git = "https://github.com/kvark/spirv_cross", branch = "wgpu5" }
 minidump_writer_linux = { git = "https://github.com/msirringhaus/minidump_writer_linux.git", rev = "029ac0d54b237f27dc7d8d4e51bc0fb076e5e852" }
+# The following overrides point to dummy projects, as a temporary measure until this is resolved:
+# https://github.com/rust-lang/cargo/issues/6179
+js-sys = { git = "https://github.com/kvark/dummy-web" }
+slotmap = { git = "https://github.com/kvark/dummy-web" }
+wasm-bindgen = { git = "https://github.com/kvark/dummy-web" }
+web-sys = { git = "https://github.com/kvark/dummy-web" }
 
 [patch.crates-io.cranelift-codegen]
 git = "https://github.com/bytecodealliance/wasmtime"
 rev = "824fa69756523f2b6d49029fe25de94130b1f144"
 
 [patch.crates-io.cranelift-wasm]
 git = "https://github.com/bytecodealliance/wasmtime"
 rev = "824fa69756523f2b6d49029fe25de94130b1f144"